src/Components/Toolbar/SaveCharacter/SaveCharacter.tsx

```diff
@@ -64,17 +61,24 @@ const SaveCharacter = ({ setShowSaveAlert }: Props) => {
       };
 
       console.log('**** New Character for ' + userId + ' is ', newCharacter);
-      const newCharacterRef = await addNewCharacterForUser(userId, newCharacter);
-      console.log('**** NewCharacterRef: ', newCharacterRef);
+      const newCharacterRef: DocumentReference<DocumentData> = await addNewCharacterForUser(
+        userId,
+        newCharacter
+      );
 
       const characterId = newCharacterRef.id;
       dispatch(setId(characterId));
-      console.log('**** NewCharacterId: ', characterId);
-      await setDoc(newCharacterRef, { id: characterId }, { merge: true });
-      console.log(`**** ${characterName} has been saved`);
-
-      setIsSaving(false);
-      setShowSaveAlert(true);
+      await createCharacterDocument(newCharacterRef, characterId);
+      console.log('****CCD ran');
+      try {
```

Comment:
the try catch doesnt work here, you the calls that you're trying to catch need to be within the try block, in this case any of the calls you're awaiting

Reply:
Thanks, I have fixed taht up and my test for rejected calls and associated alerts are passing.

Reply:
Thanks, I have fixed that up and my test for rejected calls and associated alerts are passing.